### PR TITLE
NuGet Commands: Drop-Database

### LINKED ISSUE #2476: Commands: Drop Database
In the past, we've talked about having a command to delete the database (i.e. call `EnsureDeleted`). Something like...
```cmd
dnx ef database delete
```

### PR DESCRIPTION
**Pull request by bricelam**
This includes changes from #4657. Only 03f95a27246c799f7388e11fc83b3d9bd0e2e87d needs review.

cc @natemcmaster @laskoviymishka (aka Rarity)

Resolves #2476

### CODE REVIEW

File: src/dotnet-ef/DatabaseDropCommand.cs [hunk 1]
```diff
@@ -39,9 +39,10 @@ private static int Execute(string context, string startupProject, string environ
             new OperationExecutor(startupProject, environment)
                 .DropDatabase(
                     context,
-                    message =>
+                    (database, dataSource) =>
                     {
-                        Reporter.Output.WriteLine(message);
+                        Reporter.Output.WriteLine(
+                            $"Are you sure you want to drop the database '{database}' on server '{dataSource}'? (y/N)");
                         var readedKey = Console.ReadKey().KeyChar;
```
Comment (natemcmaster) [inline src/dotnet-ef/DatabaseDropCommand.cs]: shouldn't we skip read altogether this if `--force` was used?

**Comment by natemcmaster** [comment]
It looks like `--force` on dotnet-ef doesn't actually skip the prompt.

Otherwise, :shipit:

**Comment by bricelam** [comment]
Thanks. I'll do a bit bashing on this too before merging.

File: src/Microsoft.EntityFrameworkCore.Commands/tools/EntityFramework.psm1 [hunk 2]
```diff
@@ -513,6 +513,70 @@ function Enable-Migrations {
...
+            InvokeDotNetEf $dteProject database drop @options | Out-Null
+        }
+    } else {
+        InvokeOperation $dteStartupProject $Environment $dteProject  DropDatabase @{
+            contextType = $contextTypeName
+            confirmCheck = [Func[string, string, bool]]{
+                param ($database, $dataSource)
+
+                return $PSCmdlet.ShouldProcess("$database on $dataSource")
+            }
```
Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.Commands/tools/EntityFramework.psm1]: Bad new. This results in `Cannot serialize delegates over unmanaged function pointers, dynamic methods or methods outside the delegate creator's assembly.` :disappointed:

**Comment by bricelam** [comment]
Recalling. We need to rethink how the confirmation works. One idea @natemcmaster had was to get the information in a separate command (e.g. `Get-DbContext` / `dotnet ef dbcontext show --json`), then prompt, then call drop.

**Pull request closed without merging**